sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java

```diff
@@ -0,0 +1,500 @@
+                if (!query.contains(totalWordCountPlaceHolder)) {
+                    continue;
+                }
+
+                FullTextQueryStatistics fullTextQueryStatistics = statistics.getFullTextQueryStatistics().get(statisticsIndex);
+                query = query.replace(totalWordCountPlaceHolder, Long.toString(fullTextQueryStatistics.getTotalWordCount()));
+
+                String hit_counts_array = "[" +
+                    fullTextQueryStatistics.getHitCounts()
+                        .stream()
+                        .map(Object::toString)
+                        .collect(Collectors.joining(","))
+                    + "]";
+
+                query = query.replace(hitCountsArrayPlaceHolder, hit_counts_array);
+                ++statisticsIndex;
+            }
+            return query;
+        });
+    }
+
+    private GlobalFullTextSearchQueryStatistics aggregateStatistics(List<GlobalFullTextSearchQueryStatistics> globalFullTextSearchQueryStatistics) {
+        GlobalFullTextSearchQueryStatistics aggregatedStats = new GlobalFullTextSearchQueryStatistics();
+        aggregatedStats.setDocumentCount(0L);
+        List<FullTextQueryStatistics> aggregateFullTextQueryStatistics = new ArrayList<>();
+
+        for (GlobalFullTextSearchQueryStatistics statistics : globalFullTextSearchQueryStatistics) {
+            aggregatedStats.setDocumentCount(aggregatedStats.getDocumentCount() + statistics.getDocumentCount());
+            if (aggregateFullTextQueryStatistics.isEmpty()) {
+                aggregateFullTextQueryStatistics = statistics.getFullTextQueryStatistics();
+            } else {
+                assert statistics.getFullTextQueryStatistics().size() == aggregateFullTextQueryStatistics.size();
+                for (int i = 0; i < statistics.getFullTextQueryStatistics().size(); i++) {
+                    assert statistics.getFullTextQueryStatistics().get(i).getHitCounts().size() == aggregateFullTextQueryStatistics.get(i).getHitCounts().size();
+                    aggregateFullTextQueryStatistics.get(i).setTotalWordCount(aggregateFullTextQueryStatistics.get(i).getTotalWordCount() + statistics.getFullTextQueryStatistics().get(i).getTotalWordCount());
+                    for (int j = 0; j < statistics.getFullTextQueryStatistics().get(i).getHitCounts().size(); j++) {
+                        aggregateFullTextQueryStatistics.get(i).getHitCounts().set(j, aggregateFullTextQueryStatistics.get(i).getHitCounts().get(j) + statistics.getFullTextQueryStatistics().get(i).getHitCounts().get(j));
+                    }
+                }
+            }
+            aggregatedStats.setFullTextQueryStatistics(aggregateFullTextQueryStatistics);
+        }
+        return aggregatedStats;
+    }
+
+    public static class ScoreTuple {
+        private final Double score;
+        private final Integer index;
+
+        public ScoreTuple(Double score, Integer index) {
+            this.score = score;
```

Comment:
When would scopre and index ever be null?

Reply:
These values will never be null. I have added a null check in the ctor.